docs/design/data_model.md

```diff
@@ -0,0 +1,412 @@
+
+The _InitZomesComplete_ is required to guarantee that the application's initialisation steps are only run once. This is because, to Holochain, initialisation actions are
+indistinguishable from agent actions taken at any other time.
+
+#### Creating, updating and deleting data
+
+The _Create_, _Update_, and _Delete_ actions provide create, update and delete semantics for entries in Holochain.
+
+##### Create
+
+The _Create_ action introduces new entry data. The type of that entry data defines how it should be interpreted. Entry types are described later.
+
+##### Update
+
+The _Update_ action declares that a new entry should semantically replace a previous entry. Updates do not modify the original entry or action — they create a new entry and an action that references the original.
+
+Updates establish a relationship between the old and new versions of data. Applications can query for all updates to a given _Create_ action, allowing them to construct update histories or assign meaning to the "current" version of that data.
+
+Holochain does not implicitly describe how applications should process updates.
+
+##### Delete
+
+The _Delete_ action marks an action as deleted by referencing it. Like updates, deletes do not remove data — they add metadata indicating that the entry should be considered deleted.
+
+A _Delete_ action must reference either a _Create_ or an _Update_ action.
+
+Holochain does not implicitly describe how applications should process deletes.
+
+#### Linking data
+
+The _CreateLink_ and _DeleteLink_ actions provide a mechanism for creating directed, typed relationships between entities. Links allow applications to build graph structures, indexes, and navigable relationships without embedding references directly in entry data.
+
+##### _CreateLink_
+
+The _CreateLink_ action creates a directed link from a base hash to a target hash. These hashes may be of any type, including hashes of actions, entries, agent public keys or even custom hashes. Links are first-class data in Holochain, stored on the agent's source chain as an action.
+
+The base hash is how links are grouped, so that all links from the same base hash, with the same type, are considered a set.
+
+##### _DeleteLink_
+
+The _DeleteLink_ action marks a previously created link as deleted. Like entry deletes, link deletes do not remove the original _CreateLink_ action — they add metadata indicating the link should no longer be considered active.
+
+Unlike for entries, Holochain should automatically consider *DeleteLink_s when fetching links. It should still be possible to retrieve the set of create and delete link actions so that the application can decide what the current set should be.
```

Comment:
```suggestion
Unlike for entries, Holochain should automatically consider _DeleteLink_s when fetching links. It should still be possible to retrieve the set of create and delete link actions so that the application can decide what the current set should be.
```